WordPress/gutenberg
# Button: Add hover style to `secondary` variant

![button](https://github.com/user-attachments/assets/1be7a509-2cf5-49ff-84f3-b9f9e6988c85)


## What?
Update the appearance of `secondary` variant of the `Button` component.

* Adds a clearer hover style.
  * Text color and border color are slightly darker.
  * A subtle background has been added (the same as `tertiary` buttons for now).
* Increased font weight.

## Why?
This is outlined and discussed in https://github.com/WordPress/gutenberg/issues/64744. Relevant pieces:

### Hover style
> Users with low vision might have difficulty distinguishing buttons from static content. A distinct hover style helps these users identify buttons as interactive elements.

### Increased font weight
For folks with certain visual impairments the current appearance is very similar to inputs. A small increase in font weight helps distinguish one from the other.

`-webkit-font-smoothing: antialiased` is applied to assist with legibility. This may be a separate consideration, and it may make sense to create a mixin so that it gets applied in all instances of medium font weight. Let's discuss that.

Note: We might also consider applying this to primary/tertiary buttons too, if there's a concern about consistency within the family of `Button` variants.

## To test
In Storybook:

1. `npm run storybook:dev`
2. Browse the `secondary` variant of the `Button` component
3. Observe the updated hover styles and increased font weight

In the Editor:

* Open the Site Editor.
* Observe that instances of Secondary buttons include the updated styles:
  * 'Browse all patterns' button in the Inserter.
  * 'Move to trash' in the page inspector.
  * 'Apply globally' in the block inspector.

<img width="1303" alt="Screenshot 2024-11-26 at 17 29 27" src="https://github.com/user-attachments/assets/f53c2abe-0370-41b3-b6e2-e00d63229d38">
